frontend/src/components/object/ObjectFilters.vue

```diff
@@ -0,0 +1,171 @@
+      }
+    }
+  }
+);
+
+// Computed
+const metadataValues = computed(() => {
+  // Filter out any tags that don't have an objectID that exist in getUnfilteredObjectIds
+  const filteredVals = getMetadataSearchResults.value.filter((searchRes) =>
+    getUnfilteredObjectIds.value.some((obj) => obj === searchRes.objectId)
+  );
+
+  return filteredVals
+    // Take the metadata for the objects, and flatten them into a single array
+    .flatMap((obj) => obj.metadata)
+    // Add a display property to each tag to be used by the multiselect
+    .map((val) => ({ ...val, display: `${val.key}:${val.value}` }))
+    // Unique by display property
+    .filter(
+      (val, index, self) =>
+        index === self.findIndex((t) => t.display === val.display)
+    )
+    .sort((a, b) => a.display.localeCompare(b.display));
+});
+
+const tagsetValues = computed(() => {
+  // Filter out any tags that don't have an objectID that exist in getUnfilteredObjectIds
+  const filteredVals = getTagSearchResults.value.filter((searchRes) =>
+    getUnfilteredObjectIds.value.some((obj) => obj === searchRes.objectId)
+  );
+
+  return filteredVals
+    // Take the tags for the objects, and flatten them into a single array
+    .flatMap((obj) => obj.tagset)
+    // Add a display property to each tag to be used by the multiselect
+    .map((val) => ({ ...val, display: `${val.key}=${val.value}` }))
+    // coms-id not allowed as a tag to filter on by COMS
+    .filter((val) => val.key !== 'coms-id')
+    // Unique by display property
+    .filter(
+      (val, index, self) =>
+        index === self.findIndex((t) => t.display === val.display)
+    )
+    .sort((a, b) => a.display.localeCompare(b.display));
+});
+
+// Actions
+const selectedFilterValuesChanged = () => {
+  // Get the 'display' property out from selected tag and metadata
+  const metaToSearch: Array<MetadataPair> = selectedMetadata.value.map(
+    ({ ...meta }: any) => meta
+  );
+  const tagSetToSearch: Array<Tag> = selectedTags.value.map(
+    ({ ...tag }: any) => tag
+  );
+
+  // Search the object store with the tagset as a param and metadata as headers
+  objectStore.fetchObjects(
+    {
+      bucketId: props.bucketId,
+      permCode: Permissions.READ
+    },
+    tagSetToSearch,
+    metaToSearch
+  );
+};
+
+const searchMetadata = async () => {
+  searching.value = true;
+  //await 4 seconds
+  await metadataStore.searchMetadata();
+  searching.value = false;
+};
+
+const searchTagging = async () => {
+  searching.value = true;
+  //await 4 seconds
+  await tagStore.searchTagging();
+  searching.value = false;
+};
+
+</script>
+
+<template>
+  <MultiSelect
+    v-model="selectedMetadata"
+    :options="metadataValues"
+    :loading="searching"
+    class="filterDropdown mr-4"
+    display="chip"
+    option-label="display"
+    placeholder="Metadata"
+    filter
+    filter-placeholder="Search metadata"
+    :empty-message="searching ? 'Loading...' : 'No available options'"
```

Comment:
i wonder if the 'Loading..' text should only show when the list is empty. the flickering is a bit off-putting.
maybe by setting this to `:empty-message="searching && metadataValues.length ===0 ? 'Loading...' : 'No available options'"`

Reply:
oh and not an issue for me but if we wanted to make the multiselect search field just have the bottom border (as in mockup).. i think the `.p-multiselect-filter .p-inputtext` element can be messed around with.. but not sure if there's a clean way .. or if we should

Reply:
The flicker was because before anything was calculated on showing the dropdown the current v-model state would render. Switching to `before-show` is the key to that fix